homeassistant/components/insteon/__init__.py

```diff
@@ -78,6 +80,11 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         return True
 
     conf = config[DOMAIN]
+    hass.data[DOMAIN] = {}
```

Comment:
Move this above the if-check above so you can always assume `hass.data[DOMAIN]` exists.